File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> res = new HashMap<String, List<String>>();
        for(String word: strs){
            char[] charWord = word.toCharArray();
            Arrays.sort(charWord);
            String sortedWord = new String(charWord);
            res.putIfAbsent(sortedWord, new ArrayList());
            res.get(sortedWord).add(word);
        }
        return new ArrayList(res.values());
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        encoded_strs = ""
        for word in strs:
            encoded_strs += str(len(word)) + "#" + word
        return encoded_strs

    def decode(self, s: str) -> List[str]:
        res = []
        i = 0
        j = 0
        while j < len(s):
            if s[j] == "#":
                length = int(s[i:j])
                i = j + 1
                j += length
                res.append(s[i:j+1])
                i = j + 1
                j = i + 1
            else:
                j += 1

        return res


        

